### ISSUE TITLE
Inappropriate substitution of $DIR and $WORKSPACE

### ISSUE THREAD

**Issue by aganders3** [dtolnay/trybuild#134]
I'm trying to run tests on https://github.com/PyO3/pyo3 and running into some issues with output mismatch. This is with `trybuild = "=1.0.49"`. Note that the URL `https://github.com/PyO3/pyo3/issues/1632` is replaced with `https://github.com$DIR$WORKSPACE/issues/1632`.

```
EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Python functions cannot have generic type parameters
 --> tests/ui/invalid_pyfunctions.rs:4:21
  |
4 | fn generic_function<T>(value: T) {}
  |                     ^

error: Python functions cannot have `impl Trait` arguments
 --> tests/ui/invalid_pyfunctions.rs:7:36
  |
7 | fn impl_trait_function(impl_trait: impl AsRef<PyAny>) {}
  |                                    ^^^^

error: `async fn` is not yet supported for Python functions.

Additional crates such as `pyo3-asyncio` can be used to integrate async Rust and Python. For more information, see https://github.com/PyO3/pyo3/issues/1632
  --> tests/ui/invalid_pyfunctions.rs:10:1
   |
10 | async fn async_function() {}
   | ^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Python functions cannot have generic type parameters
 --> tests/ui/invalid_pyfunctions.rs:4:21
  |
4 | fn generic_function<T>(value: T) {}
  |                     ^

error: Python functions cannot have `impl Trait` arguments
 --> tests/ui/invalid_pyfunctions.rs:7:36
  |
7 | fn impl_trait_function(impl_trait: impl AsRef<PyAny>) {}
  |                                    ^^^^

error: `async fn` is not yet supported for Python functions.

Additional crates such as `pyo3-asyncio` can be used to integrate async Rust and Python. For more information, see https://github.com$DIR$WORKSPACE/issues/1632
  --> tests/ui/invalid_pyfunctions.rs:10:1
   |
10 | async fn async_function() {}
   | ^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

**Comment by aganders3** [comment]
Maybe a solution for this would be to recognize the `<protocol>://` in the URL and skip over it?

**Comment by aganders3** [comment]
Ooops this is a result of me developing in a Docker container with the code mounted at `/pyo3` - confirmed by making a test for this repo with a similar URL and the code mounted at `/trybuild`. I can fix that test by mounting to a different location that does not conflict with the URL like that.

**Comment by dtolnay** [comment]
I published a fix in 1.0.50 so that it should work even at `/pyo3`.
